appserver/orb/orb-iiop/src/main/java/org/glassfish/enterprise/iiop/impl/IIOPSSLSocketFactory.java

```diff
@@ -651,34 +618,82 @@ private Socket createSSLSocket(String host, int port)
             _logger.log(Level.FINE, "Merged socket ciphers: " + cList);
         }
 
-        return (String[])cList.toArray(new String[cList.size()]);
+        return (String[]) cList.toArray(new String[cList.size()]);
     }
 
     /**
      * Check whether given cipherInfo belongs to given protocol.
+     *
      * @param cipherInfo
      * @param ssl2Enabled
      * @param ssl3Enabled
      * @param tlsEnabled
      */
-    private boolean isValidProtocolCipher(CipherInfo cipherInfo,
-            boolean ssl2Enabled, boolean ssl3Enabled, boolean tlsEnabled, boolean tlsEnabled11, boolean tlsEnabled12) {
+    private boolean isValidProtocolCipher(CipherInfo cipherInfo, boolean ssl2Enabled, boolean ssl3Enabled,
+            boolean tlsEnabled, boolean tlsEnabled11, boolean tlsEnabled12) {
         return (tlsEnabled && cipherInfo.isTLS() ||
                 tlsEnabled11 && cipherInfo.isTLS() ||
                 tlsEnabled12 && cipherInfo.isTLS() ||
                 ssl3Enabled && cipherInfo.isSSL3() ||
                 ssl2Enabled && cipherInfo.isSSL2());
     }
 
+    /**
+     * Checks whether SO_KEEPALIVE should be enabled on a Socket and enables it if it should be. Checks for the
+     * presence of a property on the IIOP listener and globally, preferring the value set in the listener.
+     *
+     * @param socket The socket to potentially enable SO_KEEPALIVE on
+     * @throws SocketException If there was an error enabling SO_KEEPALIVE on the socket.
+     */
+    private void enableSOKeepAliveAsRequired(Socket socket) throws SocketException {
+        boolean shouldSet = false;
+
+        // For each listener, find one with a matching port
+        for (IiopListener iiopListener : IIOPUtils.getInstance().getIiopService().getIiopListener()) {
+            if (Integer.valueOf(iiopListener.getPort()) == socket.getPort()) {
+                // Check for the property globally before checking on the specific listener
+                if (Boolean.getBoolean(SO_KEEPALIVE)) {
+                    // If it's present globally, check if we've actually set the property on the listener
+                    if (iiopListener.getPropertyValue(SO_KEEPALIVE) != null) {
```

Comment:
This if() statement isn't necessary
Boolean.getBoolean(null) will always return false, null check isn't necessary